### PR TITLE
[NFC][Clang] Fix test constexpr-function-recovery-crash.cpp

### PR DESCRIPTION
**Pull request by yronglin**

### CODE REVIEW

File: clang/test/SemaCXX/constexpr-function-recovery-crash.cpp [hunk 1]
```diff
@@ -99,6 +99,10 @@ TEST_EVALUATE(DoWhileCond, do {} while (some_cond < 10););    // expected-error
                                                               // expected-error {{constexpr variable 'forceEvaluateDoWhileCond' must be initialized by a constant expression}}
 TEST_EVALUATE(If, if (!!){};);                // expected-error + {{}}
 TEST_EVALUATE(IfInit, if (auto x = !!; 1){};);// expected-error + {{}}
-TEST_EVALUATE(ForInit, if (!!;;){};);         // expected-error + {{}}
```
Comment (yronglin) [inline clang/test/SemaCXX/constexpr-function-recovery-crash.cpp:102]: I think this case was use to check `for` statement but not `if`? Apologies if there is any mistake in my understanding.

**Comment by cor3ntin** [comment]
This looks correct to me, but I'd like @sam-mccall to confirm that it was unintended.
Maybe we should keep the old code around as loops tend to share _some_ code and we don't want `if(!!;;)` to crash either

File: clang/test/SemaCXX/constexpr-function-recovery-crash.cpp (hunk 1 above)
Comment (sam-mccall) [reply clang/test/SemaCXX/constexpr-function-recovery-crash.cpp:102]: oops, yes, the intent was to check For here

**Review by sam-mccall — APPROVED** [review]
LGTM

I don't have a strong opinion about keeping the old even-more-broken versions. They're not very likely, and we don't attempt to cover most things like that. A fuzzer is probably a better approach (we have one, it finds an ~infinite number of crashes).

**Comment by yronglin** [comment]
Thanks a lot for your review! @sam-mccall @cor3ntin

**Pull request merged**